epoxy-processor/src/main/java/com/airbnb/epoxy/processor/GeneratedModelInfo.kt

```diff
@@ -0,0 +1,279 @@
+package com.airbnb.epoxy.processor
+
+import com.airbnb.epoxy.ModelView
+import com.airbnb.epoxy.processor.Utils.buildEpoxyException
+import com.airbnb.epoxy.processor.Utils.isSubtypeOfType
+import com.squareup.javapoet.AnnotationSpec
+import com.squareup.javapoet.ClassName
+import com.squareup.javapoet.ParameterSpec
+import com.squareup.javapoet.TypeName
+import com.squareup.javapoet.TypeVariableName
+import java.util.ArrayList
+import java.util.LinkedHashSet
+import javax.lang.model.element.Element
+import javax.lang.model.element.Modifier
+import javax.lang.model.element.TypeElement
+import javax.lang.model.element.VariableElement
+
+abstract class GeneratedModelInfo(val memoizer: Memoizer) {
+
+    lateinit var superClassElement: TypeElement
+    lateinit var superClassName: TypeName
+    lateinit var parameterizedGeneratedName: TypeName
+    lateinit var generatedName: ClassName
+
+    /**
+     * Get the object type this model is typed with.
+     */
+    var modelType: TypeName? = null
+        protected set
+    var shouldGenerateModel = false
+
+    /**
+     * If true, any layout classes that exist that are prefixed by the default layout are included in
+     * the generated model as other layout options via a generated method for each alternate layout.
+     */
+    var includeOtherLayoutOptions = false
+
+    val attributeInfo: MutableList<AttributeInfo> = mutableListOf()
+    val typeVariableNames: MutableList<TypeVariableName> = ArrayList()
+    val constructors: MutableList<ConstructorInfo> = ArrayList()
+    val methodsReturningClassType: MutableSet<MethodInfo> = LinkedHashSet()
+    val attributeGroups: MutableList<AttributeGroup> = ArrayList()
+    private val attributeToGroup = mutableMapOf<AttributeInfo, AttributeGroup>()
+    val annotations: MutableList<AnnotationSpec> = ArrayList()
+
+    /**
+     * The info for the style builder if this is a model view annotated with @Styleable. Null
+     * otherwise.
+     */
+    var styleBuilderInfo: ParisStyleAttributeInfo? = null
+        private set
+
+    /**
+     * An option set via [com.airbnb.epoxy.ModelView.autoLayout] to have Epoxy create the
+     * view programmatically
+     * instead of via xml layout resource inflation.
+     */
+    var layoutParams = ModelView.Size.NONE
+
+    /**
+     * The elements that influence the generation of this model.
+     * eg base model class for @EpoxyModelClass, view class for @ModelView, etc
+     */
+    open fun originatingElements(): List<Element> {
```

Comment:
Similar - might be nice to keep this private and call into an abstract `originatingElements` methods